src/data_processing/wiki_to_netflix.py

```diff
@@ -0,0 +1,134 @@
+                                mwapi:search "%(Title)s" ;
+                                mwapi:language "en" .
+                ?item wikibase:apiOutputItem mwapi:item .
+            }
+
+            ?item wdt:P31/wdt:P279* wd:Q11424 .
+            
+            {
+                # Get US release date
+                ?item p:P577 ?releaseDateStatement .
+                ?releaseDateStatement ps:P577 ?releaseDate .
+                ?releaseDateStatement pq:P291 wd:Q30 .  
+            }
+            UNION
+            {
+                # Get unspecified release date
+                ?item p:P577 ?releaseDateStatement .
+                ?releaseDateStatement ps:P577 ?releaseDate .
+                FILTER NOT EXISTS { ?releaseDateStatement pq:P291 ?country }
+            }
+        
+            FILTER (YEAR(?releaseDate) = %(Year)d) .
+
+            ?item rdfs:label ?itemLabel .
+            FILTER (lang(?itemLabel) = "en") .
+
+            OPTIONAL {
+                ?item wdt:P136 ?genre .
+                ?genre rdfs:label ?genreLabel .
+                FILTER (lang(?genreLabel) = "en") .
+            }
+
+            OPTIONAL {?item wdt:P57 ?director.
+                            ?director rdfs:label ?directorLabel.
+                            FILTER (lang(?directorLabel) = "en")}
+
+            SERVICE wikibase:label { bd:serviceParam wikibase:language "en" . }
+            }
+    
+        ''' % {'Title': row[2], 'Year': int(row[1])}
+
+        response = requests.post('https://query.wikidata.org/sparql',
+                    headers={'User-Agent': user_agent},
+                    data={
+                    'query': SPARQL,
+                    'format': 'json',
+                    }
+        )
+        response.raise_for_status() 
+        
+        data = response.json()
+        
+        wiki_movie_ids.append(wiki_feature_info(data, 'item'))
+        wiki_genres.append(wiki_feature_info(data, 'genreLabel'))
+        wiki_directors.append(wiki_feature_info(data, 'directorLabel'))
+    
+    return(wiki_movie_ids, wiki_genres, wiki_directors)
+
+# Calling all functions
+def process_data(test=False):
+    missing_count = 0
+    processed_data = []
+
+    netflix_file = read_netflix_txt(os.path.join(base_dir, 'netflix_movies.txt'), test)
+    num_rows = len(netflix_file)
+
+    netflix_csv = os.path.join(base_dir, 'netflix_movies.csv')
+
+    wiki_movie_ids_list, wiki_genres_list, wiki_directors_list = wiki_query(netflix_file, user_agent)
+
+    for index, row in enumerate(netflix_file):
+        netflix_id, year, title = row[0], int(row[1]), row[2]
+        if wiki_movie_ids_list[index] == 'NA':
+            missing_count += 1
+        else:
+            processed_data.append([netflix_id, wiki_movie_ids_list[index], title, year, wiki_genres_list[index], wiki_directors_list[index]])
+
+    create_netflix_csv(netflix_csv, processed_data)
+
+    print('missing:', missing_count, '(', missing_count / num_rows * 100, '%)')
```

Comment:
You can do this all with `f''` strings if you want. You can put arithmetic expressions in `f''` strings:

```
print(f'found: {num_rows - missing_count}')

Reply:
See suggestion below